src/Automatonymous/AutomatonymousStateMachine.cs

```diff
@@ -533,13 +677,43 @@ protected virtual void SubState(Expression<Func<State>> propertyExpression, Stat
             SetState(name, state);
         }
 
+        protected internal virtual State<TInstance> SubState(string name, State superState)
+        {
+            if (superState == null)
+                throw new ArgumentNullException(nameof(superState));
+
+            State<TInstance> superStateInstance = GetState(superState.Name);
+
+            // If the state was already defined, don't define it again
+            if (TryGetState(name, out var existingState) &&
+                name.Equals(existingState?.Name) &&
+                superState.Name.Equals(existingState?.SuperState?.Name))
+                return existingState;
+
+            var state = new StateMachineState<TInstance>(this, name, _eventObservers, superStateInstance);
+
+            SetState(name, state);
+            return state;
+        }
+
+        protected internal virtual void SubState(State<TInstance> subState, State<TInstance> superState)
+        {
+            if (superState == null)
+                throw new ArgumentNullException(nameof(superState));
+
+            State<TInstance> superStateInstance = GetState(superState.Name);
+
+            // If the state was already defined, don't define it again
+            superStateInstance.AddSubstate(subState);

```

Comment:
Why is this needed? If the state is already created, it should have the superState already from the constructor. It isn't adding it to a list, Add is confusing, and this method seems unnecessary.

Reply:
I'm open to suggestions here. The problem came in about defining sub-state when States were already defined. The approach could be changed so that the declaration of sub-state, creates the new state instead.

Right now, this is how to use it with the builder pattern
```csharp 
.Build(builder => builder
  .State("SubState", out State<TInstance> SubState)
  .State("SuperState", out State<TInstance> SuperState)
  .SubState(SubState, SuperState)
);
```

However, maybe this is a better approach:
```csharp
.Build(builder => builder
  .State("SuperState", out State SuperState)
  .SubState("SubState", SuperState, out State SubState)
);
```

This may be better, but would rather you make that call. Thoughts?

Reply:
@phatboyg I took a look at this again. I already had this other method created. I just removed the other way of creating sub-states from the builder pattern. Not sure it adds enough value. Let me know if you feel differently.